src/usage/search.md

```diff
@@ -1,83 +1,136 @@
+Therefore you need to increase `vectors.hnsw_ef_search` until filtered vectors are sufficient.
 
-If your queries include a `WHERE` clause, you can set set search mode to `vbase`. It's good and it even works on all conditions. `vbase` is a **postfilter** method: it pulls rows as many as you need, but it scans rows that you may not need. Since some rows will definitely be removed by the `WHERE` clause, we can skip scanning them, which will make the search faster. We call it **prefilter**.
+The appropriate value depends on the input data distribution and filtering rate.
+Too large `vectors.hnsw_ef_search` will result in wasted memory.
 
-Prefilter speeds your query in the following condition:
+It's recommended in these situations:
 
-* You create a multicolumn vector index containing a vector column and many payload columns.
-* The `WHERE` clause in a query is just simple like `(id % 2 = 0) AND (age >  50)`.
+* Search without filter and transaction
+* Returning insufficient vectors is acceptable
+* The `vbase` search mode fails due to out-of-memory
 
-Prefilter is also used in internal implementation for handling deleted rows in `pgvecto.rs`.
+### `vbase`
 
-Prefilter may have a negative impact on precision. Test the precision before using it.
+`vbase` is the recommended search mode when any filter is enabled.
 
-Prefilter is enabled by default because it almost only works if you create a multicolumn vector index.
+In this mode, the filter is applied after a batch of vectors is returned.
+It will continuously request the next batch until the collected vectors are sufficient.
 
-## Options
+In most cases, `vbase` mode would return enough vectors for your filter.
+For how it works, see the thesis [VBASE: Unifying Online Vector Similarity Search and Relational Queries via Relaxed Monotonicity](https://www.usenix.org/conference/osdi23/presentation/zhang-qianxi).
 
-Search options are specified by PostgreSQL GUC. You can use `SET` command to apply these options in session or `SET LOCAL` command to apply these options in transaction.
+It's recommended in these situations:
 
-Runtime parameters for planning a query:
+* Search with filter or transaction
+* Returning sufficient vectors is important
+* Tired of tuning `vectors.hnsw_ef_search` in `basic` mode
 
-| Option               | Type    | Range              | Default   | Description                                                                  |
-| -------------------- | ------- | ------------------ | --------- | ---------------------------------------------------------------------------- |
-| vectors.enable_index | boolean |                    | `on`      | Enables or disables the query planner's use of vector index-scan plan types. |
-| vectors.search_mode  | enum    | `"basic", "vbase"` | `"basic"` | Search mode.                                                                 |
+You can enable `vbase` by `SET vectors.search_mode = vbase;`.
+
+## Prefilter
```

Comment:
We need to add info about multicolumn indexes. Users will read this documentation and find no where to use prefiltering.

Reply:
They are not implemented now, it would be confusing if we merged future and exist feature.

Reply:
cc @VoVAllen